src/cs1/linked-lists/linked-list-basics.md

```diff
@@ -0,0 +1,401 @@
+
+Here we insert `7` to the end of the list with our first method.
+
+We will create a temporary `node*` that will allow us to find the end of the list. Consider what happens if we do not create a temporary pointer. If we use the head to traverse the rest of the list then we will loose the nodes we pass over. This lose of unfreed memory is called a `memory leak`.
+
+Instead, we will create a temporary pointer called `tempHead` which we will use to traverse until the end of the list is reached.
+
+```          
+
++------+
+| head |
++------+
++------+    +------+    +-----+    +-----+    +------+
+|  10  | -> |  15  | -> |  3  | -> |  1  | -> | NULL |
++------+    +------+    +-----+    +-----+    +------+
+    ^
+    |
++----------+
+| tempHead |
++----------+
+```                  
+
+We then iterate tempHeap until the node we are at has `NULL` next node. Then, we can add the new node to our list. 
+
+```
++-----+       +------+
+|  1  | -     | NULL |     1's next node is NULL
++-----+  |    +------+
+         |       ^------
+         |    +-----+  |
+         |--> |  7  | --
+              +-----+
+```
+
+Once we've identified the last node in the list we simply add our new node to the back of the list. Since we have to traverse the WHOLE list to find the end, this makes this approach take `O(n)` time.
+
+Another way we can insert to back is by having another pointer to the end of the list. Typically we call this node `tail`. Keeping track of `tail` allows us to insert in `O(1)` time.
+
+Let's look at the `O(n)` insert code
+
+```c
+node* addToBack(node* head, int val){
+    node* tail = makeNode(val);
+    tail->data = val;
+
+    node* temp = head;
+    //If the head is NULL, then our new node is the head
+    if(temp == NULL)
+        return tail;
+    //Iterate to the end of the list
+    while(temp->next !=NULL)
+        temp = temp->next;
+
+    temp->next = tail;
+    tail->next = NULL;
+
+    //return the head of the new list
+    return head;
+}
+```
+
+Can you come up with the insert using `tail`?
+
+## Insert in Order 
+
+Besides adding either to the front of the back of a list we might want to add items in order. This will require us to add items in the middle of the list. 
+
+This approach will require us to iterate until we find the correct spot to place our new node at.
+
+Let's insert `13` into our list.
+
+```
++------+
+| head |
++------+
++------+    +------+    +------+    +------+    +------+
+|  10  | -> |  11  | -> |  15  | -> |  18  | -> | NULL |
++------+    +------+    +------+    +------+    +------+
+```
+
+What we will need to do is traverse the list until we find the first node that is greater than the value we want to add into our list. Just like add to back, we will need a temporary node in order to traverse over the list.
```

Comment:
Just more semantics, but in this case for tracking, only a pointer is made instead of a node, so I would make that clear.

```suggestion
What we will need to do is traverse the list until we find the first node that is greater than the value we want to add into our list. Just like add to back, we will need a temporary pointer in order to traverse over the list.
```